### ISSUE TITLE
Add entry `-p sct_deepseg`

### ISSUE THREAD

**Issue by jcohenadad** [spinalcordtoolbox/spinalcordtoolbox#4545]
However, given that there are multiple models in deepseg, this feature will not be straightforward. It will be part of the major refactoring. 

I'm just opening the issue for prosperity, because of https://github.com/sct-pipeline/spine-park/issues/39

**Comment by joshuacwnewton** [comment]
> However, given that there are multiple models in deepseg, this feature will not be straightforward.

Actually, the `sct_deepseg` QC that I added in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4446 does indeed try to be "general" (supporting spinal cord AND lesion seg/softseg). It works for everything but rootlets QC (https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4461). So, we are 90% there already for the `sct_deepseg` QC feature.

The [reasons](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4446/files#r1577979078) I didn't add `-p sct_deepseg` to `sct_qc` at the time were:

> 1. `sct_deepseg` QC was generated using the "new" `qc2.py` interface, and I didn't want to mix `qc2.py` and `qc.py` in our QC-specific script, as it would end up being quite messy/complex.
> 2. I figure that if folks are using `sct_qc` to generate reports (rather than the built-in per-CLI QC), then they're skilled enough to be able to re-use the existing `-p sct_deepseg_sc` or `-p sct_deepseg_lesion` to generate whichever reports they desire. 

I see now from context that you need `-p sct_qc` [for manual SC segmentations](https://github.com/sct-pipeline/spine-park/blob/8e66fb444aa1a887babaf2fc2d56b966acbe801e/batch_processing.sh#L80-L102). Since this is just a SC QC, are you able to re-use `sct_deepseg_sc` for now?

**Comment by jcohenadad** [comment]
> Since this is just a SC QC, are you able to re-use sct_deepseg_sc for now?

yup, that's exactly what I did https://github.com/sct-pipeline/spine-park/commit/46499f4fa7e7ad2bf6663763736af9863461cc58
thank you for the detailed background information!

**Comment by joshuacwnewton** [comment]
A user on the forum has requested the ability to generate the `sct_deepseg` report via `sct_qc`, so I'm increasing the priority of this issue, especially as it is long overdue.
